task1.js

```diff
@@ -0,0 +1,116 @@
+export class EventEmitter {
+  listeners = {};  // key-value pair
+
+  addListener(eventName, fn) {
+    if (Array.isArray(this.listeners[eventName])) {
+      this.listeners[eventName].push({fn: fn, isOnce: false});
+    } else {
+      this.listeners[eventName] = [{fn: fn, isOnce: false}];
+    }
+  }
+
+  on(eventName, fn) {
+    if (Array.isArray(this.listeners[eventName])) {
+      this.listeners[eventName].push({fn: fn, isOnce: false});
+    } else {
+      this.listeners[eventName] = [{fn: fn, isOnce: false}];
+    }
```

Comment:
This code dulicates implementation of `addListener`. You may use one of the method inside of the other to avoid code duplication.